### DISCUSSION TITLE
Dynamic table name?

### DISCUSSION THREAD

**Discussion by etztrefis** [prisma/prisma#9534] [Q&A] [answered]
Is there a way to dynamically pass the table name?

I've tried something like this: 
```js
await prisma.$queryRawUnsafe(
    `SELECT * FROM ? WHERE SenderID = ?;`,
    req.query.table,
    req.query.user,
  );
```
and
```js
await prisma.$queryRaw(
    `SELECT * FROM ${req.query.table} WHERE SenderID = ${req.query.user};`
  );
```
But then I've noticed that the **documentation** says: 
- Note: You cannot pass a table or column name as a variable into a parameterized query. For example, you cannot SELECT ? and pass in * or id, name based on some condition.

As you might see, the table name dynamically comes from the request query. 
So I can't use something like:
 ```js
await prisma.table_name.findMany();
```

Any thoughts?

**Comment by ryands17** [comment, marked as answer]
@etztrefis 👋 

With the template string approach, it's not possible to pass the table names dynamically but you can do it using `$queryRawUnsafe` like this:

```ts
const result = await prisma.$queryRawUnsafe(`SELECT * FROM ${req.query.table}`)
```

This will work but you are responsible for sanitising the input beforehand.

You can also use: 

```ts
await prisma.[req.query.table].findMany();
```

But you won't be able to get type definitions for the response types and such as this becomes dynamic then.

**Reply by etztrefis** [reply]
Thanks for the answer! 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
